### PR TITLE
Added support for updating post status.

### PR DESCRIPTION
**Pull request by scyt**
The idea for this is a choice field with the different post status as available choices.

This choice field can be populated via GPPA:

### CODE REVIEW

File: gravity-forms/gw-update-posts.php [hunk 1]
```diff
@@ -65,6 +66,10 @@ public function update_post_by_entry( $entry, $form ) {
 			$post->post_author = (int) rgar( $entry, $this->_args['author'] );
 		}
 
+		if ( $this->_args['status'] ) {
+			$post->post_status = rgar( $entry, $this->_args['status'] );
+		}
```
Comment (spivurno) [inline gravity-forms/gw-update-posts.php:71]: It'd be awesome if we could get a list of [registered post statuses](https://developer.wordpress.org/reference/functions/get_post_stati/) and then make sure that the user-selected (or enetered) status is a valid post status before we attempt to set it. If not, we could just skip updating it.

**Review by spivurno — CHANGES_REQUESTED** [review]
I dig it! Great summary as well. Just one reliability note below.

**Comment by scyt** [comment]
Good call. I added a check to see if the post status exists. If it doesn't, it skips updating.

**Review by spivurno — APPROVED** [review]
Perfect!

**Pull request merged**